### PR TITLE
Container + Screen Widgets

### PR DESCRIPTION
**Pull request by shedaniel**
Some heavy class renaming in `BeaconScreen` so tell me if it is bad lol

### CODE REVIEW

File: mappings/net/minecraft/client/gui/container/BeaconScreen.mapping [hunk 1]
```diff
@@ -1,31 +1,60 @@
...
+		FIELD b v I
+		METHOD <init> (IIII)V
+			ARG 1 x
+			ARG 2 y
+			ARG 3 u
+			ARG 4 v
 	FIELD k BG_TEX Lqs;
-	FIELD l doneButton Ldaq$b;
-	FIELD m canConsumeGem Z
+	FIELD l buttonDone Ldaq$b;
```
Comment (NeunEinser) [inline mappings/net/minecraft/client/gui/container/BeaconScreen.mapping]: Suffixing is preferred afaik, So why not `doneButton`?

File: mappings/net/minecraft/client/gui/widget/EntryListWidget.mapping [hunk 2]
```diff
@@ -5,6 +5,8 @@ CLASS cwm net/minecraft/client/gui/widget/EntryListWidget
 			ARG 3 mouseY
 		METHOD render (IIIIIIIZF)V
 			ARG 1 index
+			ARG 2 y
+			ARG 3 x
```
Comment (NeunEinser) [inline mappings/net/minecraft/client/gui/widget/EntryListWidget.mapping:9]: Is this the right order? Usually it's x, y.

File: mappings/net/minecraft/client/gui/container/BeaconScreen.mapping (hunk 1 above)
Comment (shedaniel) [reply mappings/net/minecraft/client/gui/container/BeaconScreen.mapping]: wait why did i do that, reverting the change

File: mappings/net/minecraft/client/gui/widget/GameOptionSliderWidget.mapping [hunk 3]
```diff
@@ -1,5 +1,8 @@
 CLASS cxf net/minecraft/client/gui/widget/GameOptionSliderWidget
 	FIELD a option Lcvq;
+	METHOD <init> (Lcvo;IIIILcvq;)V
+		ARG 1 options
+		ARG 6 option
```
Comment (NeunEinser) [inline mappings/net/minecraft/client/gui/widget/GameOptionSliderWidget.mapping:5]: The difference between the two is not immediately clear to me.

File: mappings/net/minecraft/client/gui/widget/EntryListWidget.mapping (hunk 2 above)
Comment (shedaniel) [reply mappings/net/minecraft/client/gui/widget/EntryListWidget.mapping:9]: Yes blame mojang

File: mappings/net/minecraft/client/gui/widget/GameOptionSliderWidget.mapping (hunk 3 above)
Comment (shedaniel) [reply mappings/net/minecraft/client/gui/widget/GameOptionSliderWidget.mapping:5]: options is the instance of GameOptions, option is the actual option itself.
Not sure what I should name them

File: mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping [hunk 4]
```diff
@@ -49,7 +51,12 @@ CLASS cww net/minecraft/client/gui/widget/TextFieldWidget
 		ARG 1 cursor
 	METHOD g getCursor ()I
 	METHOD g setMaxLength (I)V
+		ARG 1 maximumLength
+	METHOD h setEditableColor (I)V
+		ARG 1 color
 	METHOD i isVisible ()Z
+	METHOD i setUneditableColor (I)V
+		ARG 1 color
```
Comment (NeunEinser) [inline mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping:59]: A setter for something you cannot edit? What exactly is `uneditableColor` and `editableColor`?

File: mappings/net/minecraft/client/gui/widget/EntryListWidget.mapping (hunk 2 above)
Comment (NeunEinser) [reply mappings/net/minecraft/client/gui/widget/EntryListWidget.mapping:9]: Okay, then.

File: mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping (hunk 4 above)
Comment (shedaniel) [reply mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping:59]: the text color of the textfieldwidget 
For example, it uses the editableColor if it is editable.

File: mappings/net/minecraft/client/gui/widget/GameOptionSliderWidget.mapping (hunk 3 above)
Comment (NeunEinser) [reply mappings/net/minecraft/client/gui/widget/GameOptionSliderWidget.mapping:5]: Though it's more verbose, `gameOptions` should point out the difference better. Not sure if there might be a better solution.

**Review by modmuss50 — APPROVED** [review]
Seems alright to me.

**Review by asiekierka — APPROVED** [review]

**Pull request merged**